### ISSUE TITLE
Removing labels once plot is made?

### ISSUE THREAD

**Issue by dataquestions** [datastorm-open/visNetwork#268]
Currently have this code: 

```
nodes <- data.frame(id = 1:3, label = letters[1:3])

edges <- data.frame(from = c(1,2), to = c(1,3))

# Define UI
ui <- shinyUI(fluidPage(
  visNetworkOutput("network"), 
  actionButton("removelabels", "Remove letters from nodes!")
))

# Define server
server <- shinyServer(function(input, output) {
  output$network <- renderVisNetwork({
    visNetwork(nodes, edges) %>% visOptions(highlightNearest = TRUE) %>%
      visPhysics(enabled = FALSE) %>% visEdges(smooth = FALSE)
  })
  
})
  # format positions
 
# Run the application 
shinyApp(ui = ui, server = server)
```
I would like to find a way in R that when the graph is created if i click the action button then the labels would be removed from the graph - but the graph itself would not be re run at every instance - so i want something like visNetworkProxy("network") %>% visNodes(labels = NULL) - however my labels are within my nodes dataframe and i cannot change this!

Any ideas welcome for help 

tl;dr 

Remove labels once graph is created - but not re run the graph all together - using visNetworkProxy or something

**Comment by dataquestions** [comment]
@bthieurmel any help?

**Comment by Dmirijs-Ozernovs** [comment]
Hi!
I've had similar issue. Solution was to set font size to 1 or 0 - labels remain but become invisible.
Not very elegant solution but for my purposes it was fine.

**Comment by dataquestions** [comment]
Dmirjis thank you for your help

Do you know how i can disable physics from being on the plot after i render it? 

I want it so that the network is produced but does not mode when a node is pulled - by disabling physics i can move only one node at a time?

Any ideas would be great - i would not want to use a button on a shiny app rather an observe?

**Comment by Dmirijs-Ozernovs** [comment]
I don't know is it possible to disable physics after you've created a graph with enabled physics. But if you use physics to get a nice layout - there are two options:
1. Try to experiment with igraph layouts, maybe one of them will produce acceptable results even with disables physics.
2. Create two buttons, one for fixing nodes positions (pressed after all nodes are stable), the second is for recreating the new graph with disabled physics with coordinates saved from initial graph.

About dragging nodes - you can select multiple nodes () and drag them even with disables physics, these options are not related [link](https://github.com/datastorm-open/visNetwork/issues/232).

**Comment by dataquestions** [comment]
Thank you dmirjis 

I will definitely look into this 

I have disabled physics through the visnetworkproxy %>% visPhysics (enabled = FALSE) but it runs through an action button on my shiny app 

Which is not great as i would want it to stop physics straight after rendering

I was planning to use visSetOptions or visRedraw but found no examples online

**Comment by dataquestions** [comment]
I currently am using the layout_nicely algorithm but with physics enabled and have used in a list different values for the repulsion parameter - its made the network so great but physics exist!!!

**Comment by bthieurmel** [comment]
You can use ``visEvents`` like this : 

``

nodes <- data.frame(id = 1:50, group=sample(1:10, 50, replace=T))
edges <- data.frame(from = sample(1:50, 200, replace=T), to = sample(1:50, 200, replace=T))

# if you want to see the stabilization before
visNetwork(nodes, edges) %>% visPhysics(stabilization = FALSE) %>%
  visEvents(stabilized="function () {this.setOptions( { physics: false } );}")

# or not
visNetwork(nodes, edges) %>% 
  visEvents(stabilizationIterationsDone="function () {this.setOptions( { physics: false } );}")
``

**Comment by Dmirijs-Ozernovs** [comment]
bthieurmel, thanks a lot!
This works fine and I should definitely look in visEvents function. Wish there would be more information or tutorials about all this functionality.
